src/components/Button/index.tsx

```diff
@@ -0,0 +1,134 @@
+import type { HTMLAttributes, ReactElement } from 'react'
+import { hexToCSSFilter } from 'hex-to-css-filter'
+import styled from '@emotion/styled'
+import type { StyledProps } from '@types'
+import type { Theme } from '@emotion/react'
+
+export const BUTTON_STYLE_KEYS = {
+  GHOST: 'ghost',
+  OUTLINE: 'outline',
+  OUTLINE_DISABLED: 'outlineDisabled',
+  SOLID_DISABLED: 'solidDisabled',
+  SOLID_PRIMARY: 'solidPrimary',
+  SOLID_SUB: 'solidSub'
+}
+const ICON_SIZE = 24
+
+type ButtonStyle = typeof BUTTON_STYLE_KEYS[keyof typeof BUTTON_STYLE_KEYS]
+type ButtonSize = 'large' | 'medium' | 'small'
+export interface ButtonProps extends HTMLAttributes<HTMLButtonElement> {
+  buttonStyle?: ButtonStyle
+  size?: ButtonSize
+  iconUrl?: string
+  children: string
+}
+type StyledButtonProps = StyledProps<ButtonProps, 'buttonStyle' | 'size'>
+
+export const Button = ({
+  size = 'medium',
+  buttonStyle = 'solidPrimary',
+  iconUrl,
+  children,
+  ...props
+}: ButtonProps): ReactElement => {
+  const isSmallSize = size === 'small'
+
+  return (
+    <StyledButton buttonStyle={buttonStyle} size={size} {...props}>
+      {!isSmallSize && iconUrl && (
+        <img alt="icon" height={ICON_SIZE} src={iconUrl} width={ICON_SIZE} />
+      )}
+      {children}
+    </StyledButton>
+  )
+}
```

Comment:
```suggestion
  const isSmallSize = size === 'small'
  const isShowIcon = !isSmallSize && iconUrl

  return (
    <StyledButton buttonStyle={buttonStyle} size={size} {...props}>
      {isShowIcon && (
        <img alt="icon" height={ICON_SIZE} src={iconUrl} width={ICON_SIZE} />
      )}
      {children}
    </StyledButton>
  )
}
```
중첩 조건을 따로 빼는 건 어떠신가요?